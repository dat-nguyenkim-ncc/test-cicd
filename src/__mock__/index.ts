import { TagGroupType } from './../types/index.d'
import {
  CompanyBusiness,
  CompanyDetail,
  CompanyFinancials,
  CompanyOverview,
  CompanyPeople,
  MappingSummary,
  TagData,
  TagDataParent,
} from '../types'
import { EnumAttachmentType, EnumCompanyTypeSector } from '../types/enums'

const getRandomId = (): string => {
  return (~~(Math.random() * 999999) + new Date().valueOf()).toString()
}

const companyTypes = Object.values(EnumCompanyTypeSector)
const suffix = ['Restaurant', 'Limited', 'UK', 'US', 'Bank']
export const mockCompanyId = getRandomId()
export const companyDetails = (): CompanyDetail => ({
  companyId: mockCompanyId,
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  countryCode: 'GB',
  countryName: 'United Kingdom',
  url: 'https://monzo.com',
  companyTypes: [companyTypes[~~(Math.random() * companyTypes.length)]],
})

export const MockMonzoDetails = companyDetails()

export const mappingSummary = [
  {
    title: 'CATEGORIES',
    items: [
      { label: 'Fintech', parent: [] },
      { label: 'Retail Accounts' },
      { label: 'Accounts & Savings' },
      { label: 'Digital Banking' },
    ],
    type: EnumCompanyTypeSector.FIN,
  },
  {
    title: 'SECTORS',
    items: [{ label: 'No sector mapped', parent: [] }],
    type: EnumCompanyTypeSector.FIN,
  },
] as MappingSummary[]

export const companySources = [
  {
    company: MockMonzoDetails,
    source: {
      label: 'Source: XPTO',
      default: true,
    },
  },
  {
    company: companyDetails(),
    source: {
      label: 'Manual',
    },
  },
]

export const companyOverview = (): CompanyOverview => ({
  id: getRandomId(),
  source: 'CRB',
  sources: ['CRB', 'DR', 'MANUAL'],
  companyName: `Monzo ${suffix[~~(Math.random() * suffix.length)]}`,
  companyLocation: [
    {
      id: '123456',
      address: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      postalCode: '94103',
      location: {
        region: 'EMEA',
        city: 'San Francisco',
        country: 'United States',
      },
      isHeadQuarter: true,
      selfDeclared: true,
      expandStatus: 'Following',
      source: 'bcg',
    },
  ],
  description: 'Monzo is a fintech challenger bank.',
  companyType: 'Startup',
  foundedYear: 2012,
  status: 'The company status',
  expandStatus: 'Following',
  expandStatusId: '1',
  closedDate: new Date(2012, 11, 1).toISOString(),
  ftse: 1000,
  logoUrl: 'https://monzo.com/logo.svg',
  contactEmail: 'monzo@monzo.com',
  phoneNumber: '0123456789',
  tags: null,
  otherNames: null,
  lastFundingType: null,
  logo_bucket_url: '',
  hashed_image: '',
  attachments: [
    {
      name: 'File nam',
      description: 'Broadwalk House<br>5 Appold Street<br>London<br>EC2A 2AG',
      type: EnumAttachmentType.FCT_PROFILE,
      url_attachment: '',
      expandStatus: 'Following',
      selfDeclared: true,
      date_created: '2021-03-30 10:43:33',
    },
  ],
  aliases: [
    {
      alias_id: '200354bcg1',
      company_alias: 'name-edit',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
    {
      alias_id: '200354bcg2',
      company_alias: 'test1',
      expand_status_id: 1,
      selfDeclared: true,
      source: 'bcg',
    },
  ],
})

export const companyFinancials = (): CompanyFinancials => ({
  valuation: {
    value: 23123123,
    currency: 'USD',
  },
  fundingTotal: {
    value: 2312312312,
    currency: 'USD',
  },
  companyStage: 'Series E+',
  lastFundingDate: {
    value: 1231233,
    currency: 'USD',
  },
  equityFundingTotal: {
    value: 1231233,
    currency: 'USD',
  },
  leadInvestor: 'Mussum Cassilds-Cepacol',
  lastFundingAmount: {
    value: 123123123,
    currency: 'USD',
  },
  fundingRounds: [
    {
      id: '1',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
        other:
          'Goodwater Capital, VC Orange, Corporate Passion Capital, VC Reference Capital (formerly Genevest), VC Stripe, Tech Thrive Capital, VC Vanderbilt University, Other Y Combinator, Accelerator',
      },
    },
    {
      id: '2',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
    {
      id: '3',
      date: new Date(2012, 11, 1).toISOString(),
      quarter: 'Q2',
      comment: 'This is comment',
      roundTypes: ['Equity Financing, series E+'],
      expandRound1: 'Equity Financing',
      originExpandRound1: 'Equity Financing',
      valuation: {
        value: 123123123,
        currency: 'USD',
      },
      investment: {
        value: 123123123,
        currency: 'USD',
      },
      investors: {
        lead: 'Accel Partners, VC',
      },
    },
  ],
  isExternalViewDetail: true,
})

export const companyBusiness = (): CompanyBusiness => ({
  productsServices: ['Offers a current account and debit card '],
  businessRevenue: ['Service type: B2C'],
  targetClients: ['Retail customers'],
  partnerships: [
    '**Deliveroo:** Partnered with Deliveroo in Sep-2017 to allow users to split the bill on orders via its Monzo.me money request feature',
  ],
  vision: ['string'],
  keyMetrics: ['string'],
  differentiators: ['string'],
})

export const companyAcquisitions = () => []
export const companyIpos = () => []
export const companyPeople = (): CompanyPeople => ({ companyId: 145803 })

const tags = [
  'Capital Markets',
  'Corporate Banking',
  'Retail Banking',
  'SME Banking',
  'Support',
  'Technology',
  'Wealth Management',
]

const getRandomValueLabel = (index?: number) => {
  const tag = index ? tags[index] : tags[~~(Math.random() * tags.length)]
  return {
    label: tag,
    value: tag.toLowerCase().replace(' ', '_'),
  }
}

const getRandomBoolean = () => {
  return Math.random() * 10 > 5
}

export const TagMock = (
  index?: number,
  level: number = 0,
  parent: TagDataParent[] = []
): TagData => {
  const { value, label } = getRandomValueLabel(index)
  const id = getRandomId()
  const rowId = getRandomId()
  const isPriority = getRandomBoolean()
  return {
    id,
    rowId,
    value,
    label,
    isPriority,
    endpoint: level > 0 ? Math.random() > 0.5 : false,
    parent,
    description: 'Mock',
    children:
      level >= 5
        ? undefined
        : new Array(Math.max(3, ~~(Math.random() * tags.length)))
            .fill(null)
            .map((_a, index) =>
              TagMock(index, level + 1, [...parent, { id, rowId, label, value, isPriority }])
            ),
  }
}

export const ListRandomTags = (): TagData[] =>
  new Array(tags.length).fill(null).map((_a, index) => TagMock(index, 0))

const labels = [
  'Insurance Value Chain 1',
  'Insure Tech Tag',
  'Revenue model',
  'P&C Commercial Customers',
  'Product',
  'RegTech',
  'Regulation',
  'Report',
]

export const TagGroupMock = (): TagGroupType[] =>
  new Array(labels.length * 2).fill(null).map((_a, index) => {
    const id = getRandomId()
    const rowId = getRandomId()
    const label = labels[index % labels.length]
    const value = label.toLowerCase().split(' ').join('_')
    const isPriority = getRandomBoolean()
    return {
      id,
      rowId,
      label,
      value,
      isPriority,
      parent: [],
      children: new Array(Math.max(1, ~~(Math.random() * labels.length * 2)))
        .fill(null)
        .map((_b, index) => {
          const labelChild = labels[index % labels.length]
          return {
            label: labelChild,
            id: getRandomId(),
            rowId: getRandomId(),
            parent: [{ id, label, value }],
            isPriority,
            value: labelChild.toLowerCase().split(' ').join('_'),
          }
        }),
    }
  })
