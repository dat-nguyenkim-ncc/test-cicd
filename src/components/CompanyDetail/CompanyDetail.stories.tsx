import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import CompanyDetail, { CompanyDetailProps } from './CompanyDetail'
import { Box } from 'theme-ui'
import {
  companyFinancials,
  companyOverview,
  companyPeople,
  mappingSummary,
} from '../../__mock__'

export default {
  title: 'Company Details/CompanyDetail',
  component: CompanyDetail,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta

const Template: Story<CompanyDetailProps> = args => (
  <Box sx={{ p: 7, backgroundColor: 'gray03', width: '100%' }}>
    <CompanyDetail {...args} />
  </Box>
)

export const Default = Template.bind({})
Default.args = {
  overview: companyOverview(),
  financials: companyFinancials(),
  business: companyPeople(),
  people: companyPeople(),
  mapping: mappingSummary,
} as CompanyDetailProps
