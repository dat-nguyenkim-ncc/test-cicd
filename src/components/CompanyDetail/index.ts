export { default } from './CompanyDetail'
export type { CompanyActionButton, KeysTabs, Tabs } from './CompanyDetail'
