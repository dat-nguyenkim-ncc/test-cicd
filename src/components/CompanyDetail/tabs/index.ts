export { default as Overview } from './CompanyDetailOverview'
export { default as Financials } from './CompanyDetailFinancials'
export { default as Fundraising } from './CompanyDetailFundraising'
export { default as Business } from './CompanyDetailBusiness'
export { default as People } from './CompanyDetailPeople'
export { default as Acquisitions } from './CompanyDetailAcquisitions'
export { default as Ipos } from './CompanyDetailIpos'
export { default as News } from './CompanyDetailNews'
export { default as Acquirees } from './CompanyDetailAcquirees'
export { default as UseCase } from './CompanyDetailUseCase'
export { default as Partnerships } from './CompanyDetailPartnerships'
export { default as Products } from './CompanyDetailProducts'
export { default as Investments } from './CompanyDetailInvestments'
export { default as Tractions } from './CompanyDetailTractions'
