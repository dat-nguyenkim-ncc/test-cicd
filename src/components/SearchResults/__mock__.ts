export const data = {
  external: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '7b4545e6-a34d-47c2-853e-ff9eb40c18ef',
      source: 'CRB',
      companyDetails: {
        companyId: '9185a077-c260-ac40-2678-fef9e645591b',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: 'a67f6ebd-c2e4-4c15-8657-4d40cc7626f1',
      source: 'DR',
      companyDetails: {
        companyId: '74473',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '89c20324-1390-4ea2-acbc-7cc05a98db59',
      source: 'CRB',
      companyDetails: {
        companyId: '33164743-a09a-45cb-a160-a54ace4dc3da',
        companyName: 'Monzon Biomed',
        url: 'http://www.monzonbiomed.com/',
        countryName: 'Spain',
        countryCode: 'ES',
      },
    },
    {
      id: '807a6c17-a4ff-4b74-b9ef-fee94364f451',
      source: 'CRB',
      companyDetails: {
        companyId: '456754eb-67c6-98db-780a-327db4c05c8f',
        companyName: 'monzoom.com',
        url: 'https://www.monzoom.com',
        countryName: 'Austria',
        countryCode: 'AT',
      },
    },
    {
      id: '0511b523-c25b-443b-932f-4d501d5c0cd4',
      source: 'CRB',
      companyDetails: {
        companyId: '09c52685-0f60-4a6f-a5fc-5c51ae5bec3a',
        companyName: 'Monzoon Networks',
        url: 'http://hotspot.monzoon.net',
        countryName: 'Switzerland',
        countryCode: 'CH',
      },
    },
  ],
  internal: [
    {
      //replace property id by company_id
      //TODO: change all records
      company_id: '6',
      source: 'MANUAL',
      companyDetails: {
        companyId: '6',
        companyName: 'MonzoBCG',
        url: 'www.MonzoBCG.com',
        countryName: 'England',
        countryCode: null,
      },
    },
    {
      id: '127237',
      source: 'DR',
      companyDetails: {
        companyId: '127237',
        companyName: 'Monzo',
        url: 'https://monzo.com',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
    {
      id: '141132',
      source: 'MANUAL',
      companyDetails: {
        companyId: '141132',
        companyName: 'Monzo',
        url: 'https://monzo.com/',
        countryName: 'United Kingdom',
        countryCode: 'GB',
      },
    },
  ],
}
