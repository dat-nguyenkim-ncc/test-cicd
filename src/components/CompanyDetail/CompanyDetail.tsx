import React, { useState } from 'react'
import { Box, Flex } from 'theme-ui'
import { Button, ButtonText, CompanyLogo } from '../'
import { ETLRunTimeContext } from '../../context'
import { Certification } from '../../pages/CompanyForm/CertificationForm'
import { ColumnNames } from '../../pages/CompanyForm/helpers'
import { Technology } from '../../pages/CompanyForm/TechnologyForm'
import { TechnologyProvider } from '../../pages/CompanyForm/TechnologyProvider'
import strings from '../../strings'
import {
  CompanyAcquiree,
  CompanyAcquisitionsDetail,
  CompanyFinancials,
  CompanyIposDetail,
  CompanyOverview,
  CompanyPeople,
  MappingSummary,
  TagNewsData,
  Variants,
} from '../../types'
import { ApiSourceType, EnumExpandStatus } from '../../types/enums'
import ChartDataOverView from '../ChartDataOverview/ChartDataOverView'
import { CompanySource } from '../CompanySource'
import { Heading, Paragraph, Section } from '../primitives'
import SideSummary from '../SideSummary'
import {
  Acquirees,
  Acquisitions,
  Business,
  Financials,
  Fundraising,
  Investments,
  Ipos,
  News,
  Overview,
  Partnerships,
  People,
  Products,
  Tractions,
  UseCase,
} from './tabs'
import { CompanyNewsChartResult } from './tabs/CompanyDetailNews/components'
import CompanyDetailTechnology from './tabs/CompanyDetailTechnology/CompanyDetailTechnology'

export type CompanyActionButton = {
  label: string
  variant?: Variants
  onPress(tab: string): void
}

export type CompanyDetailProps = {
  overview: CompanyOverview
  financials?: CompanyFinancials
  investments?: CompanyPeople
  fundraising?: {
    companyId: number
  }
  business?: CompanyPeople
  people?: CompanyPeople
  acquisitions?: CompanyAcquisitionsDetail
  'use-case'?: CompanyPeople
  mapping?: MappingSummary[]
  onBack(): void
  actionButton?: CompanyActionButton[]
  ipos?: CompanyIposDetail
  news?: TagNewsData
  acquirees?: CompanyAcquiree
  isExternalViewDetail?: boolean
  technology?: {
    technology: Technology[]
    technologyProvider: TechnologyProvider[]
    technologyCertification: Certification[]
  }
  partnerships?: CompanyPeople
  products?: CompanyPeople
  tractions?: CompanyPeople
  companyNewsChartRes?: CompanyNewsChartResult
  setPageCurrent(state: State): void
}

export const Tabs = {
  overview: { label: 'overview', tab: Overview },
  financials: { label: 'financials', tab: Financials },
  investments: { label: 'investments', tab: Investments },
  acquisitions: { label: 'acquisitions', tab: Acquisitions },
  acquirees: { label: 'acquirees', tab: Acquirees },
  ipos: { label: 'ipos', tab: Ipos },
  business: { label: 'business', tab: Business },
  'use-case': { label: 'use-case', tab: UseCase },
  fundraising: { label: 'fundraising', tab: Fundraising },
  technology: { label: 'technology', tab: CompanyDetailTechnology },
  people: { label: 'people', tab: People },
  news: { label: 'news', tab: News },
  partnerships: { label: 'partnerships', tab: Partnerships },
  products: { label: 'products', tab: Products },
  tractions: { label: 'tractions', tab: Tractions },
}

export type KeysTabs = keyof typeof Tabs
export type State = {
  active: KeysTabs
}

const CompanyDetail = ({
  onBack,
  actionButton,
  mapping,
  isExternalViewDetail,
  setPageCurrent,
  ...props
}: CompanyDetailProps) => {
  const [state, setState] = useState<State>({ active: 'overview' })
  const { companyDetail: copy } = strings

  const { checkTimeETL } = React.useContext(ETLRunTimeContext)

  const onClickTabs = (tab: KeysTabs) => {
    setState({ ...state, active: tab })
    setPageCurrent({ ...state, active: tab })
  }

  const buttons = Object.values(Tabs).reduce((p, c) => {
    if (props[c.label as keyof typeof props] !== undefined) {
      p.push(c.label as KeysTabs)
    }

    return p
  }, [] as KeysTabs[])

  const renderTab = () => {
    const Tab = Tabs[state.active].tab
    const data = props[state.active]

    if (!data || Tab === undefined || !Tab) return null

    return <Tab data={data as any} />
  }

  const imgSize = props.overview.expandStatus !== EnumExpandStatus.DUPLICATED ? 108 : 134

  const logoUrl =
    (isExternalViewDetail ? props?.overview.logoUrl : props?.overview.logo_bucket_url) || ''

  const isActive = state.active === ColumnNames.PRODUCTS

  return (
    <Section
      sx={{
        width: isActive ? '95vw' : '100%',
        marginLeft: isActive ? 'calc((-95vw + 1024px)/2)' : '',
        marginRight: isActive ? 'calc((-95vw + 1024px)/2)' : '',
        maxWidth: isActive ? '95vw' : '100%',
      }}
    >
      <ButtonText onPress={onBack} />
      <Flex mt={5} sx={{ justifyContent: 'space-between', flexWrap: 'wrap' }}>
        <Flex sx={{ justifyContent: 'flex-start', alignItems: 'center', gap: 3 }}>
          <CompanyLogo src={logoUrl} cursor="auto" width={imgSize} height={imgSize} />
          <Box>
            <Heading as="h1">{props.overview.companyName}</Heading>
            <CompanySource
              source={props.overview.source as ApiSourceType}
              sources={props.overview.sources as ApiSourceType[]}
            />
            <Paragraph>
              {props.overview.expandStatus === EnumExpandStatus.DUPLICATED ? copy.duplicated : ''}
            </Paragraph>
          </Box>
        </Flex>
        <Flex sx={{ flex: 1, width: '100%', justifyContent: 'flex-end' }}>
          {(actionButton || []).map((b, index) => (
            <Button
              key={index}
              sx={{ minWidth: 210, m: 3 }}
              onPress={() => {
                if (!checkTimeETL()) return
                b.onPress(state.active)
              }}
              label={b.label}
              variant={b.variant || 'primary'}
            />
          ))}
        </Flex>
      </Flex>
      <Box sx={{ overflow: 'hidden' }}>
        <Flex mt={6} sx={{ gap: 3, flexWrap: 'wrap' }}>
          {buttons.map((b, index) => (
            <Button
              key={index}
              active={state.active !== b}
              sx={{
                minWidth: state.active === b ? 180 : 'auto',
              }}
              onPress={() => onClickTabs(b)}
              variant={b === state.active ? 'primary' : 'muted'}
              label={copy.buttons[b as keyof typeof copy.buttons]}
            />
          ))}
        </Flex>
        {state.active === 'overview' && (
          <Flex
            sx={{
              bg: 'transparent',
              float: 'right',
              position: 'relative',
              maxWidth: '28%',
              mt: 6,
              flexDirection: 'column',
            }}
          >
            <ChartDataOverView companyNewsChartRes={props.companyNewsChartRes}></ChartDataOverView>
            {mapping && <SideSummary content={mapping} />}
          </Flex>
        )}

        <Box>{renderTab()}</Box>
      </Box>
    </Section>
  )
}

export default CompanyDetail
