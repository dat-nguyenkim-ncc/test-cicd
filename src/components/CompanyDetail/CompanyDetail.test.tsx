import * as React from 'react'
import { shallow } from 'enzyme'
import CompanyDetail from './CompanyDetail'
import { companyFinancials, companyOverview, companyPeople, mappingSummary } from '../../__mock__'

describe('CompanyDetail', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <CompanyDetail
        onBack={() => {}}
        mapping={mappingSummary}
        financials={companyFinancials()}
        business={companyPeople()}
        people={companyPeople()}
        overview={companyOverview()}
        setPageCurrent={() => {}}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
