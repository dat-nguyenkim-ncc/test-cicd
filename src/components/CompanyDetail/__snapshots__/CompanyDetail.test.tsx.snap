// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyDetail matches snapshot 1`] = `ShallowWrapper {}`;
